File: src/windows/magma/app/compile/value/StringNode.c
#include "./StringNode.h"
export class StringNode implements Value {
	&[I8] value;
	constructor (&[I8] value) {
		this.value = value;
	}
}

&[I8] generate(Platform platform) {
	return "\"" + this/*auto*/.value + "\"";
}
Option<Value> toValue() {
	return new Some<Value>(this/*auto*/);
}
Option<Value> findChild() {
	return new None<Value>(/*auto*/);
}
Option<&[I8]> generateAsEnumValue(&[I8] structureName, Platform platform) {
	return new None<&[I8]>(/*auto*/);
}
Type type() {
	return PrimitiveType/*auto*/.Auto;
}

File: src/node/magma/app/compile/value/StringNode.ts
import { Value } from "../../../../magma/app/compile/value/Value";
import { Platform } from "../../../../magma/app/io/Platform";
import { Option } from "../../../../magma/api/option/Option";
import { Some } from "../../../../magma/api/option/Some";
import { None } from "../../../../magma/api/option/None";
import { Type } from "../../../../magma/api/Type";
import { CompileState } from "../../../../magma/app/compile/CompileState";
import { PrimitiveType } from "../../../../magma/app/compile/type/PrimitiveType";
export class StringNode implements Value {
	value: string;
	constructor (value: string) {
		this.value = value;
	}
	generate(platform: Platform): string {
		return "\"" + this/*auto*/.value + "\"";
	}
	toValue(): Option<Value> {
		return new Some<Value>(this/*auto*/);
	}
	findChild(): Option<Value> {
		return new None<Value>(/*auto*/);
	}
	resolve(state: CompileState): Type {
		return PrimitiveType/*auto*/.Auto;
	}
	generateAsEnumValue(structureName: string, platform: Platform): Option<string> {
		return new None<string>(/*auto*/);
	}
}
